app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/DepartureView.kt

```diff
@@ -0,0 +1,185 @@
+     * Sets the line symbol name
+     *
+     * @param symbol Symbol e.g. U6, S1, T14
+     */
+    fun setSymbol(symbol: String, highlight: Boolean) {
+        val mvvSymbol = MVVSymbol(symbol)
+        symbolView.setTextColor(mvvSymbol.textColor)
+        symbolView.text = symbol
+        symbolView.backgroundTintList = ColorStateList.valueOf(mvvSymbol.backgroundColor)
+
+        if (highlight) {
+            if (big) {
+                setBackgroundColor(mvvSymbol.backgroundColor)
+                lineView.setTextColor(Color.WHITE)
+                for (index in 0 until timeSwitcher.childCount) {
+                    val tw = timeSwitcher.getChildAt(index) as TextView
+                    tw.setTextColor(Color.WHITE)
+                }
+            } else {
+                setBackgroundColor(0x20ffffff and mvvSymbol.backgroundColor)
+            }
+        } else {
+            setBackgroundColor(mvvSymbol.textColor)
+            lineView.setTextColor(Color.BLACK)
+
+            for (index in 0 until timeSwitcher.childCount) {
+                val tw = timeSwitcher.getChildAt(index) as TextView
+                tw.setTextColor(Color.GRAY)
+            }
+        }
+    }
+
+    /**
+     * Sets the line name
+     *
+     * @param line Line name e.g. Klinikum Großhadern
+     */
+    fun setLine(line: CharSequence) {
+        lineView.text = line
+    }
+
+    /**
+     * Sets the departure time
+     *
+     * @param departureTime Timestamp in milliseconds, when transport leaves
+     */
+    fun setTime(departureTime: DateTime) {
+        this.departureTime = departureTime
+        updateDepartureTime()
+    }
+
+    private fun updateDepartureTime() {
+        val departureOffset = Seconds.secondsBetween(DateTime.now(), departureTime ?: DateTime.now()).seconds
+
+        if (departureOffset > 0) {
+
+            val hours = departureOffset / ONE_HOUR_IN_SECONDS
+            val minutes = departureOffset / ONE_MINUTE_IN_SECONDS % MINUTES_PER_HOUR
+            val seconds = departureOffset % ONE_MINUTE_IN_SECONDS
+
+            val text = if (hours > 0) {
+                String.format(Locale.getDefault(), "%2d:%02d:%02d", hours, minutes, seconds)
+            } else {
+                String.format(Locale.getDefault(), "%2d:%02d", minutes, seconds)
+            }
+
+            timeSwitcher.setCurrentText(text)
+        } else {
+            animateOut()
+            return
+        }
+        // Keep countDown approximately in sync.
+        mHandler.postDelayed({ updateDepartureTime() }, 1000)
+    }
+
+    private fun animateOut() {
+        valueAnimator = ValueAnimator.ofFloat(0.0f, 1.0f)
+                .setDuration(500).apply {
+                    addUpdateListener(SlideOutAnimator())
+                    interpolator = AccelerateInterpolator()
+                    start()
+                }
+    }
+
+    /**
+     * Call this, when the DepartureView isn't needed anymore.
+     */
+    fun removeAllCallbacksAndMessages() {
```

Comment:
This shouldn’t be part of the public API, but be automatically executed when the parent activity is destroyed. You can use a custom `LifecycleObserver` for that (see [example](https://proandroiddev.com/make-your-custom-view-lifecycle-aware-its-a-piece-of-cake-90b7c0498686)).